zebra/zserv.c

```diff
@@ -182,13 +182,20 @@ static void zserv_encode_interface(struct stream *s, struct interface *ifp)
 static void zserv_encode_vrf(struct stream *s, struct zebra_vrf *zvrf)
 {
 	struct vrf_data data;
+	const char *ns_name = zvrf_ns_name(zvrf);
 
 	data.l.table_id = zvrf->table_id;
+
+	if (ns_name)
+		strncpy(data.ns_name, basename(ns_name), NS_NAMSIZ);
+	else
+		memset(data.ns_name, 0, NS_NAMSIZ);
 	/* Pass the tableid */
 	stream_put(s, &data, sizeof(struct vrf_data));
 	/* Interface information. */
 	stream_put(s, zvrf_name(zvrf), VRF_NAMSIZ);
-
+	/* Pass the fd from NSID */
+	stream_put(s, &data, sizeof(struct vrf_data));
```

Comment:
We already sent the whole `vrf_data` structure before, no need to do it again.

Reply:
ok